File: app/src/main/java/io/weicools/purereader/data/DailyGankData.java
package io.weicools.purereader.data;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Create by weicools on 2018/4/15.
 * <p>
 * desc: 每日干货Data
 */

public class DailyGankData {
    @SerializedName("error")
    @Expose
    private boolean error;
    @SerializedName("category")
    @Expose
    private List<String> category;

    @SerializedName("results")
    @Expose
    public Results results;

    public boolean getError() {
        return error;
    }

    public void setError(boolean error) {
        this.error = error;
    }

    public List<String> getCategory() {
        return category;
    }

    public void setCategory(List<String> category) {
        this.category = category;
    }

    public Results getResults() {
        return results;
    }

    public void setResults(Results results) {
        this.results = results;
    }

    public static class Results {
        @SerializedName("Android")
        @Expose
        public List<GankData> androidList;

        @SerializedName("iOS")
        @Expose
        public List<GankData> iOSList;

        @SerializedName("前端")
        @Expose
        public List<GankData> webFontList;

        @SerializedName("App")
        @Expose
        public List<GankData> appList;

        @SerializedName("拓展资源")
        @Expose
        public List<GankData> resourceList;

        @SerializedName("瞎推荐")
        @Expose
        public List<GankData> recommendList;

        @SerializedName("休息视频")
        @Expose
        public List<GankData> videoList;

        @SerializedName("福利")
        @Expose
        public List<GankData> girlList;
    }
}

File: app/src/main/java/io/weicools/purereader/data/DailyGankData.kt
package io.weicools.purereader.data

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * @author Weicools Create on 2018.08.12
 *
 * desc:
 */
data class DailyGankData(
    @SerializedName("category") @Expose val category: List<String>,
    @SerializedName("error") @Expose val error: Boolean,
    @SerializedName("results") @Expose val results: Results
)

data class Results(
    @SerializedName("Android") @Expose val Android: List<GankContent>,
    @SerializedName("App") @Expose val App: List<GankContent>,
    @SerializedName("iOS") @Expose val iOS: List<GankContent>,
    @SerializedName("休息视频") @Expose val video: List<GankContent>,
    @SerializedName("前端") @Expose val webFont: List<GankContent>,
    @SerializedName("拓展资源") @Expose val resource: List<GankContent>,
    @SerializedName("瞎推荐") @Expose val recommend: List<GankContent>,
    @SerializedName("福利") @Expose val welfare: List<GankContent>
)

/*
https://gank.io/api/day/2018/08/09

{
  "category": [
    "前端",
    "iOS",
    "瞎推荐",
    "App",
    "拓展资源",
    "Android",
    "福利",
    "休息视频"
  ],
  "error": false,
  "results": {
    "Android": [],
    "App": [],
    "iOS": [],
    "休息视频": [],
    "前端": [],
    "拓展资源": [],
    "瞎推荐": [],
    "福利": []
  }
}
 */